### PR TITLE
FEAT: Support Applymap Operand

### LINKED ISSUE #413: FEAT: Support applymap op
Ref: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.applymap.html

### PR DESCRIPTION
**Pull request by Hank0626**
## What do these changes do?
- Support dataframe applymap operand 
- add unittest

## Related issue number
Fixes #413 

## Check code requirements
- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/dataframe/base/tests/test_applymap_execution.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import pandas as pd
+import pytest
+
+from .... import dataframe as md
+
+
+def test_frame_applymap_execution(setup):
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/tests/test_applymap_execution.py:21]: Better to move these tests into `test_base_execution.py`.

File: python/xorbits/_mars/dataframe/base/applymap.py [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+            )
+        new_nsplits = list(in_df.nsplits)
+        new_op = op.copy()
+        kw = out_df.params.copy()
+        new_nsplits = tuple(new_nsplits)
+        kw.update(dict(chunks=chunks, nsplits=new_nsplits))
+        return new_op.new_tileables(op.inputs, **kw)
+
+    @classmethod
+    def tile(cls, op):
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/applymap.py:106]: Move code into this function, no need to create another classmethod `_tile_df`.

File: python/xorbits/_mars/dataframe/base/tests/test_applymap_execution.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    df = pd.DataFrame([[1, 2.12], [3.356, 4.567]])
+    mdf = md.DataFrame(df)
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/tests/test_applymap_execution.py:23]: Add some cases to test if it works with multiple chunks. For example,
```
df = pd.DataFrame(np.random.rand(10, 2))
mdf = md.DataFrame(df, chunk_size=3)
```

**Review by aresnow1 — COMMENTED** [review]
Seems something wrong with coverage, see if it still fail for next push.

File: python/xorbits/_mars/dataframe/base/applymap.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    build_df,
+    build_empty_df,
+    make_dtype,
+    make_dtypes,
+    parse_index,
+    validate_output_types,
+)
+
+
+class ApplymapOperand(DataFrameOperand, DataFrameOperandMixin):
```
[the remark this answers is not in the document]
Comment (ChengjieLi28) [reply python/xorbits/_mars/dataframe/base/applymap.py:36]: > I mock the `ApplyOperand`. So can I also change `ApplyOperand` to `DataFrameApply`?

``Apply`` method is available in both ``series`` and ``df``. So I think naming ``DataFrameApplymap`` here is more suitable. What do you think?

File: python/xorbits/_mars/dataframe/base/applymap.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
+# Copyright 2022-2023 XProbe Inc.
+# derived from copyright 1999-2021 Alibaba Group Holding Ltd.
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/base/applymap.py:2]: For new files, this line can be removed.

File: python/xorbits/_mars/dataframe/base/applymap.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+    func,
+    na_action=None,
+    args=(),
+    dtypes=None,
+    dtype=None,
+    name=None,
+    output_type=None,
+    index=None,
+    skip_infer=False,
+    **kwds,
```
Comment (ChengjieLi28) [inline python/xorbits/_mars/dataframe/base/applymap.py:204]: Add type annoations for these parameters. And check other places when you define a func.

File: python/xorbits/_mars/dataframe/base/tests/test_base_execution.py [hunk 7]
```diff
@@ -542,6 +542,135 @@ def test_apply_with_arrow_dtype_execution(setup):
     pd.testing.assert_series_equal(result, expected)
 
 
+def test_data_frame_applymap_execute(setup):
```
Comment (ChengjieLi28) [inline python/xorbits/_mars/dataframe/base/tests/test_base_execution.py:545]: You can add a TODO here to support GPU for ``applymap`` op when you have time and gpu machine.

File: python/xorbits/_mars/dataframe/base/tests/test_base_execution.py [hunk 8]
```diff
@@ -542,6 +542,135 @@ def test_apply_with_arrow_dtype_execution(setup):
...
+
+    # test empty dataframe
+    df_raw = pd.DataFrame()
+    df = from_pandas_df(df_raw, chunk_size=2)
+
+    r = df.applymap(lambda x: len(str(x)))
+    result = r.execute().fetch()
+    expected = df_raw.applymap(lambda x: len(str(x)))
+    pd.testing.assert_frame_equal(result, expected)
+
```
Comment (ChengjieLi28) [inline python/xorbits/_mars/dataframe/base/tests/test_base_execution.py:672]: Nit: add a test to test the func passed to the ``applymap`` that have args.

**Review by ChengjieLi28 — APPROVED** [review]

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**